### DISCUSSION TITLE
Something wrong...

### DISCUSSION THREAD

**Discussion by SMFDrummer** [HemulGM/ChatGPT#42] [Show and tell]
This is definitely the worst Android ChatGPT I have ever used. The UI always crashes and the feel is extremely bad. I often talk nonsense and never answer my questions. I filled in my own api and correctly set the GPT-4 model, but it still shows GPT-3. Can you use Dim sum for your software?

**Comment by HemulGM** [comment]
There are general settings and settings for each individual chat. Each chat can be customized in its own way. Before writing something like this, I suggest thinking with your own head so as not to look like you are now.

**Comment by HemulGM** [comment]
Show what is there and where it crashes.

**Comment by SMFDrummer** [comment]
_I think this is a problem that you can do well, or you may not be proficient in Pascal, or Pascal cannot be the first choice for developing Android applications. If so, I strongly recommend and call for you to develop and abandon Pascal in the UI interface! What I know is that compared to Android applications developed by Java or Kotlin, it may not be as good as Java or Kotlin in terms of performance. In addition, Google does not recommend using this language to develop Android applications. In the increasingly higher SDK environment, the Pascal, which is not independent of Google, will become worse and worse._
**The first question is that** when entering anything in my input box, if I did not trigger the return of the system, but chose to sink my input method, the software of the software will be stuck in the half -screen. If I close the software at this time, I close the software at this time The next startup of the software will directly retreat, and the second start will return to normal. I suggest that you modify your UI's listener about the return of the monitoring system and change the software to read the UI recorder or recovery device. They must have problems.
**The second question is that** when I click the setting of the settings interface, the input method will pop up, but what I clicks is only an indescribable description. I suggest that you modify the set of settings description, which will use TextView instead of EditText.
**The third question is that** when I set my API correctly and choose the GPT-4 model correctly, the software call call of the software is still GPT-3 during use. I suggest you check whether the setting item is properly passed on, or whether the parameter is correctly drawn.
_Since I am not proficient in Pascal, I can't give constructive suggestions through your code. This is all the bugs I have temporarily discovered. If you can listen to my suggestion, you can understand that I scold you without any reason. I also hope that you will give up the Pascal language in software UI and switch to Google's Kotlin+Jetpack Compose or Java+XML. **People will always make progress. You must learn to follow up new things instead of sticking to Delphi. This language is only suitable for scientific research and calculations. He should live in this field and not to maliciously expand his scope of use._**

**Reply by HemulGM** [reply]
1. You are mistaken in Delphi language. This language was developed specifically for the commercial development of large-scale applications. Delphi is not Pascal, it is his descendant. At the same time, Pascal was also not developed for the scientific field, this is a lie. It was designed for people not related to science, it was designed more simply for this. So that people can study it without getting a degree. Delphi uses the same SDK as Java/Kotlin and compiles native app. However, the UI in it is NOT native. It is cross-platform and will look the same on all platforms.
2. Your description of the problems is extremely bad and not clear. It would be better if you record a video. And also, specify the version of Android. Because I and hundreds of other people use the application on Android and have no problems.
Once again about the model. The "Settings" button opens the GENERAL settings and it says that there are settings for NEW chats, and not for already created chats. Each chat has its own settings button at the bottom.
I created this application in 2 weeks, sitting for a couple of hours in the evenings. How long will you be doing it in Kotlin or Java?

Yes, it doesn't have smooth animations because the framework is currently limited to 30fps on Android. This will be corrected soon.

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-19-35-47-334_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/521b2c73-deb4-4d1b-8ea5-8fe899e91528)

![Screenshot_2023-07-29-19-35-52-269_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/e14a3054-d9e9-4ff4-8413-0ea2196cb03c)

![Screenshot_2023-07-29-19-35-41-872_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/79bfacec-6d13-4506-824b-1049dab52feb)

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-19-45-32-812_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/d7595811-9948-4eec-8ca8-48470be4b1d4)

![Screenshot_2023-07-29-19-43-56-030_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/f5a543e2-437c-429f-8a7c-ede385ff76d7)

![Screenshot_2023-07-29-19-44-36-427_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/7a49a25d-4bce-499f-b62e-e3335fa24462)

**Reply by HemulGM** [reply]
![image](https://github.com/HemulGM/ChatGPT/assets/22197242/6867a7a9-f0b8-423b-9900-71804ee3504a)

**Reply by SMFDrummer** [reply]
The software UI card is half screen, the UI stack is not properly processed, and the instructions for setting items such as organization will pop up input methods, nonsense, and incorrect GPT model when clicked. I think my description is quite clear

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-19-39-39-787_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/c8139fc6-371c-4210-85e3-fe9f215bdf6a)

**Reply by SMFDrummer** [reply]
Did you see that the settings interface is stuck in the middle and even intertwined with the navigation view

**Reply by HemulGM** [reply]
What version of Android do you have? It is obvious that it does not work correctly. It does not notify about hiding the keyboard. On modern, new phones, everything works correctly.

**Reply by SMFDrummer** [reply]
I use the Qualcomm Snapdragon 865 processor, with a screen resolution of 2340x1080 (6.67 inch full screen), Android 13, corresponding to SDK 33, and Google Security Update 2023.03.01

**Reply by SMFDrummer** [reply]
Now Google has generally upgraded to Android 13, and the Android studio I am using has reached SDK 34, which is Android 14 preview. The recommended minimum SDK compatibility for the model is 24, in other words, the SDK I can use is 24-33

**Reply by SMFDrummer** [reply]
But I think as long as I can open it, it's not a problem with the SDK. You should carefully check the stack of the software UI, and now it seems that you haven't handled the UI conflicts properly

**Reply by HemulGM** [reply]
https://youtu.be/2092iOX4Wmw

**Reply by SMFDrummer** [reply]
Because you are using a Google keyboard, it does not actively lower the button, but my input method has. You listen to the system return to make the software UI follow the input method rebound, but I cannot. As long as my input method retracts, the software UI will get stuck in half because I did not execute the return key, and if I execute the return key in half of the position, the software will exit directly

**Reply by HemulGM** [reply]
The system has clear events for opening or closing the keyboard. The system sends an event with the keyboard size and the application makes an offset of the active screen by the keyboard size. The hide keyboard event removes offsets. What is not clear here? This event is generated by the system. Why the system does not generate it is another question.

**Reply by HemulGM** [reply]
What keyboard are you using? And where are the other problems you wrote about? Or is it the only one? And why is this a "bad app"?

**Reply by SMFDrummer** [reply]
I use the Chinese input methods. My other questions are in the screenshot above. One is nonsense. I say hi, hello, and he says a lot of things I don't know what they are. I adjust the GPT4 model, which is actually GPT3

**Reply by SMFDrummer** [reply]
I have searched through the entire GitHub and found that only your application supports GPT4 Android applications. Since China is not a legal usage area for GPT, I have modified the baseURL to include forwarding and correct APIs within China. What I can confirm is that this will not affect the program, as all forwarding is consistent with the official OpenAI

**Reply by HemulGM** [reply]
Are you sure you're a developer? Your reaction to all this is not very clear.
1. This is the official API! If a specific model is selected in the chat settings, it is used in the API request. It works correctly from the very beginning of the project. No one but you complained about it. Manually specify a non-existent model and you will see an error from the server, not from the program. Because I don't control it. I gave full options for working with the API. I don't care what model you specify, it will be in the next request.
2. If the model answered this way, then the API answered that way. The program does not use any connections other than accessing the OpenAI API and the content is not modified or processed in any way. You work directly with their official API

**Reply by HemulGM** [reply]
Now I have an account without GPT-4, but I had an account with access to this model and I tested it. Everything worked correctly and regularly. So these are questions for your "intermediaries", and not for my application.

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-20-23-58-238_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/176f0188-00d0-40d4-b0ee-59ff5784c776)

**Reply by HemulGM** [reply]
THIS IS GENERAL SETTINGS! EVERY CHAT HAS ITS OWN SETTINGS!

**Reply by HemulGM** [reply]
![image](https://github.com/HemulGM/ChatGPT/assets/22197242/186447aa-b62d-4767-abc0-1103a0115f51)

**Reply by SMFDrummer** [reply]
hold on, let me try

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-20-27-56-895_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/2b306a54-cab8-44ca-b180-b90e2b962737)

![Screenshot_2023-07-29-20-27-49-979_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/de91ec77-6362-4509-84dd-82394b30412f)

**Reply by SMFDrummer** [reply]
What's wrong bro, why?

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-20-29-25-851_mark.via.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/3ef468af-b257-4838-bd74-d7e3bec79961)

that's true which is on my website, using the same api and the same baseurl, this website also apply for opanai official api.

**Reply by HemulGM** [reply]
All questions to your intermediary. He is probably not very honest, because with the official API everything works as it should.

**Reply by HemulGM** [reply]
OK. Let's do it. Specify a non-existent model there (in the chat settings) and see what happens

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-20-33-51-948_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/42337daf-cc8a-4e82-ab09-4ca3f3e852d5)

![Screenshot_2023-07-29-20-33-58-096_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/febfb28c-ae45-4c16-badd-dd3ce0160013)

**Reply by SMFDrummer** [reply]
OK.What question do you have?

**Reply by HemulGM** [reply]
Do you see that the model is being passed to the API? And the API told you that there is no such model. So your model is also transmitted and its answers depend only on your intermediary, and not on me.

**Reply by SMFDrummer** [reply]
My intermediary does not have the Poe model, so I was prompted when I passed it on, which is normal. You asked me to test it. And just now, when this model was still GPT-4, my mediation passed it correctly, but it returned GPT-3, which proves that my mediation has the GPT4 model, but your program access is still fed back to me through GPT3

**Reply by SMFDrummer** [reply]
That's the reality, and I tested using a site that works the same as your software. The configuration is exactly the same, and your software output is incorrect

**Reply by HemulGM** [reply]
Do you understand that "gpt-4" is TRANSMITTED in the request? A query is built and your model is passed with the chat history for context! There is nothing extra to work with the GPT-4 model! It works exactly the same as with the GPT-3 and other models. The desired model is simply passed in the request!

**Reply by SMFDrummer** [reply]
And what about that nonsense? You don't have to blame my agent too, do you? I've already used nearly $1000 in credit and never had such a problem before. Why did I start talking nonsense when I came to you

**Reply by HemulGM** [reply]
Do you understand that "gpt-4" is TRANSMITTED in the request? A query is built and your model is passed with the chat history for context! There is nothing extra to work with the GPT-4 model! It works exactly the same as with the GPT-3 and other models. The desired model is simply passed in the request!

**Reply by SMFDrummer** [reply]
> And what about that nonsense? You don't have to blame my agent too, do you? I've already used nearly $1000 in credit and never had such a problem before. Why did I use your application，but it talking nonsense

**Reply by HemulGM** [reply]
Open Postman. And make a request directly for verification. Since you are a developer, it will not be difficult. It's very simple. Make sure that it is not my application that deceives you, but your intermediary.

**Reply by SMFDrummer** [reply]
Okay, I don't know what happened right now, and I don't know how to adjust your software settings. It's working properly now. I checked my API backend and your software can now send GPT4 and GPT4-0613 (32k) messages correctly. Everything is confusing, I don't know how to adjust the settings, and I don't know why it suddenly improved, The previous models sent in the background are indeed based on GPT3.5, but it suddenly changed

**Reply by SMFDrummer** [reply]
Now that's it, besides my own customer service input method issue, it's the UI logic of your software. I have screenshots for you to refer to, which is the only place I can confirm there is a problem right now

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-19-39-39-787_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/ace15915-a791-48a2-a04d-0914b44f2761)

![Screenshot_2023-07-29-19-35-52-269_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/30d22fc2-51b6-4389-b8c9-d8d18659c2b3)

![Screenshot_2023-07-29-19-35-47-334_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/1e34780a-b014-4e7c-ba31-6b40c856183b)

**Reply by HemulGM** [reply]
Here the questions are apparently to the keyboard. Try another, just for the sake of testing. Because other users do not have this problem. I also have Android 13, but on MagicOS 7.1.
How to test with your keyboard, I don't know. If you tell me the name of the keyboard, I will try to deal with the problem.

**Reply by SMFDrummer** [reply]
this keyboard named sougou input, official website is https://shouji.sogou.com/wap/
if you cannot open this, please tell me   immediately

**Reply by SMFDrummer** [reply]
press the 立即下载 button then it will download

**Reply by SMFDrummer** [reply]
If you want to use Russian input, you can click on the top left corner of the keyboard, drop it down to the bottom layer, select more settings, select a multilingual keyboard, and then drop it down. There is a Russian keyboard in the list, and click the blue add button on the right to add it

**Reply by SMFDrummer** [reply]
![Screenshot_2023-07-29-21-35-44-344_com.embarcadero.ChatGPT.jpg](https://github.com/HemulGM/ChatGPT/assets/61400112/3eecf4e9-85e1-470e-85c3-af390332a4a6)

no being this bro..... please...please... What's wrong with you...

**Reply by HemulGM** [reply]
This is the response from the server

**Reply by HemulGM** [reply]
Post it here and I'll check it out. Don't be so stupid yourself.

**Reply by SMFDrummer** [reply]
ah？

**Reply by SMFDrummer** [reply]
ridiculous...

**Reply by SMFDrummer** [reply]
this sentence is 我有一个java项目，包含了两个方法，一个是获取传参，一个是对传参的处理，我能确认的是，获取传参需要一些时间，这会不会导致对传参的处理方法运行时使用的是已经失效的传参而不是新的传参
translated
I have a Java project that includes two methods: one is to obtain parameters, and the other is to process parameters. What I can confirm is that obtaining parameters takes some time, which will cause the processing method for parameters to run using expired parameters instead of new ones

**Reply by HemulGM** [reply]
![Screenshot_20230729_184704.jpg](https://github.com/HemulGM/ChatGPT/assets/22197242/ae1ec11a-833b-4495-a5e3-86c348a8e029)

**Reply by HemulGM** [reply]
Everything works correctly with the official API!!!

**Reply by SMFDrummer** [reply]
Okay, my intermediary may not really be compatible with your application, as it sometimes returns correct and sometimes doesn't work. I'm sorry, but I have to do this. It's very difficult to recharge GPTplus in China and there are restrictions on the plus. I also need to use a VPN in the United States to access it, which is too difficult. Network restrictions have forced me to use a domestic forwarding station. The forwarding limit in China is: Question 0.18/1K Tokens, answer 0.36/1K Tokens, and basically sending a "Hello" will require a deduction of 0.03 RMB, equivalent to 0.4 rubles.

**Reply by HemulGM** [reply]
The application works according to the regular official API. By following and using the recommended mechanics. It can work with other APIs as long as they support the same communication protocol.
In Russia, ChatGPT is also blocked, but we have unlimited access to the official API. Are you limited?
Try to remove the intermediary.

**Reply by SMFDrummer** [reply]
Coincidentally, my API can only be used through intermediary forwarding because it is not an official API, it is an encrypted API. I am sorry, but there is nothing I can do about it

**Reply by HemulGM** [reply]
With the keyboard, I will try to solve the problem. It was partially possible to repeat.

**Reply by SMFDrummer** [reply]
Looking forward to your reply

**Comment by MrSitnikov** [comment]
Gives an error:"Unable to open the program "ChatGPT_arm64.app"" when trying to run the app on macOS arm.

**Reply by HemulGM** [reply]
https://github.com/HemulGM/ChatGPT/issues/52
